api/scripts/daily-digest.ts

```diff
@@ -0,0 +1,125 @@
+import { SMTPClient } from "emailjs";
+import { createEmailWithTemplate } from "../emails/htmlEmailUtils.js";
+import config from "@config";
+import type { MonitoringParams } from "../api/V1/monitoringPage.js";
+import { calculateMonitoringPageCriteria } from "../api/V1/monitoringPage.js";
+import { generateVisits } from "../api/V1/monitoringVisit.js";
+import modelsInit from "@models/index.js";
+import { RecordingType } from "@typedefs/api/consts.js";
+
+(async () => {
+  try {
+    const models = await modelsInit();
+    const users = await models.User.findAll({where: {
+      'settings.emailNotifications.dailyDigest': true
+    }});
+    
+    const templateFilename = "daily-digest.html";
+    const client = new SMTPClient(config.smtpDetails);
+    // const client = new SMTPClient({
+    //   user: "",
+    //   password: "",
+    //   host: "smtp.gmail.com",
+    //   ssl: true,
+    // });
+
+    users.forEach(async user => {
+      const requestUser = user;
+      const groups = await models.GroupUsers.findAll({where: {
+        'UserId': user.id
+      }});
+
+      groups.forEach(async group => {
+        const recordingData = {};
+        const name = await models.Group.findOne({where: {
+          'id': group.GroupId
+        }});
+
+        const stations = await models.Station.findAll({where: {
+          'GroupId': group.GroupId
+        }});
+        const stationsArray = [];
+        stations.forEach(async station => {
+          stationsArray.push(station.id);
+        });
+
+        const oneWeekAgo = new Date();
+        const now = new Date();
+        oneWeekAgo.setHours(oneWeekAgo.getHours() - 168);
+
+        const params: MonitoringParams = {
+          stations: stationsArray,
+          groups: [group.GroupId],
+          page: 1,
+          pageSize: 100,
+          from: oneWeekAgo,
+          until: now,
+          types: [RecordingType.ThermalRaw]
+        };
+
+        const viewAsSuperAdmin = true;
+        const searchDetails = await calculateMonitoringPageCriteria(
+          requestUser,
+          params,
+          viewAsSuperAdmin
+        );
+      
+        searchDetails.compareAi = "Master";
+        searchDetails.types = params.types;
+      
+        const visits = await generateVisits(
+          requestUser.id,
+          searchDetails,
+          viewAsSuperAdmin
+        );
+        
+        (visits as unknown as Array<any>).forEach(async visit => {
```

Comment:
I'd like to see the actual typing here, as the return value of `generateVisits` does have a well known, well defined type – which also has an error variant that should be handled.  Typing it as unknown is asking for bugs to be introduced.